### ISSUE TITLE
ZAP breaks JavaScript on pages, if content is on a long line

### ISSUE THREAD

**Issue by lol768** [zaproxy/zaproxy#8151]
### Describe the bug

ZAP has some unhelpful `LONG_CONTENT` handling which seems to forcibly insert line-breaks if breakpoints are set and the lines are considered to be "too long".

This happens regardless of the impact of inserting a linebreak, and breaks webpages where the HTML has been minified (in my case, the linebreaks had been inserted in the middle of a JS `<script>` causing a `SyntaxError`).

![Screenshot from 2023-10-29 18-27-52](https://github.com/zaproxy/zaproxy/assets/2552726/e71a0554-2e78-4c83-ac16-4084b8fad42c)

As far as I can tell, this makes it impossible to intercept and modify HTTP requests/responses with ZAP where the source application has been minified.

How can I configure/turn off this behaviour?

If the problem is the text editor, would a better option not be to have "phantom soft breaks" which exist for editor performance, but don't actually result in a modification to the response structure?

IntelliJ's editor does this, you can see it's forced onto multiple lines but these don't actually form part of the text:

![Screenshot from 2023-10-29 18-25-27](https://github.com/zaproxy/zaproxy/assets/2552726/48c3186e-e965-4045-8dad-d857f086c474)

### Steps to reproduce the behavior

1. Try to intercept a HTTP request/response involving long lines

### Expected behavior

1. User is able to intercept and modify the request/response

### Software versions

ZAP
Version: 2.14.0

Installed Add-ons: [[id=alertFilters, version=18.0.0],
[id=ascanrules, version=58.0.0], [id=authhelper,
version=0.10.0], [id=automation, version=0.33.0],
[id=bruteforce, version=15.0.0], [id=callhome,
version=0.8.0], [id=commonlib, version=1.18.0],
[id=database, version=0.3.0], [id=diff, version=14.0.0],
[id=directorylistv1, version=7.0.0], [id=domxss,
version=18.0.0], [id=encoder, version=1.4.0], [id=exim,
version=0.7.0], [id=formhandler, version=6.5.0], [id=fuzz,
version=13.12.0], [id=gettingStarted, version=16.0.0],
[id=graaljs, version=0.5.0], [id=graphql, version=0.20.0],
[id=help, version=17.0.0], [id=hud, version=0.18.0],
[id=importurls, version=8.0.0], [id=invoke, version=14.0.0],
[id=network, version=0.12.0], [id=oast, version=0.17.0],
[id=onlineMenu, version=12.0.0], [id=openapi,
version=37.0.0], [id=postman, version=0.2.0],
[id=pscanrules, version=52.0.0], [id=quickstart,
version=43.0.0], [id=replacer, version=15.0.0], [id=reports,
version=0.26.0], [id=requester, version=7.4.0], [id=retest,
version=0.8.0], [id=retire, version=0.26.0], [id=reveal,
version=7.0.0], [id=saverawmessage, version=6.0.0],
[id=savexmlmessage, version=0.2.0], [id=scripts,
version=42.0.0], [id=selenium, version=15.15.0], [id=soap,
version=20.0.0], [id=spider, version=0.7.0], [id=spiderAjax,
version=23.17.0], [id=tips, version=12.0.0],
[id=webdriverlinux, version=64.0.0], [id=websocket,
version=30.0.0], [id=zest, version=42.0.0]]

Operating System: Linux
Architecture: amd64
Java Version: Azul Systems, Inc. 17.0.5
System's Locale: en_GB
Display Locale: en_GB
Format Locale: en_GB
Default Charset: UTF-8
Look and Feel: Nimbus (javax.swing.plaf.nimbus.NimbusLookAndFeel)

### Would you like to help fix this issue?

- [ ] Yes

**Comment by thc202** [comment]
The behaviour is to prevent #5791.

**Comment by lol768** [comment]
I figured as much. I tried Pakiki and whilst it does display the full response and allow editing it, the UI just slows down to a halt.

I wonder if an interim fix might be to insert linebreaks by running the response (if it's HTML) through some sort of beautifier? You'd have a guarantee then that it doesn't _break_ the response structure.

**Comment by psiinon** [comment]
@lol768 what are you actually trying to do?
If you want to change big messages on the fly then you can use the Replacer or httpsender scripts.

**Comment by lol768** [comment]
Right, but I was trying to use the Break tab and the interactive editor there. Those are reasonable workarounds, if I know what I want to edit consistently, but it does lead to a cumbersome UX if I want to make adhoc changes.

Intercepting and editing the same response works fine in Burp Suite. I guess the real answer here is that the performance of RSyntaxTextArea isn't good enough in its current state.

**Comment by thc202** [comment]
> I don't like the workaround that was put in place around that because I don't think a tool like ZAP should ever automatically break a forwarded response.

That was not on purpose, otherwise this would have been closed already.

**Comment by psiinon** [comment]
Thats why its called a workaround rather than a solution.
However we dont really want to fix and maintain our own version of RSyntaxTextArea ..

**Comment by lol768** [comment]
>  However we dont really want to fix and maintain our own version of RSyntaxTextArea ..

Having done some digging, I'm fairly sure Burp uses the same library. So I guess they've patched it themselves.

It's a damn shame it didn't use a copyleft LICENSE.

**Comment by thc202** [comment]
They have PRed changes, no idea about fixes for this case though.
